pkg/cmd/resource/show.go

```diff
@@ -0,0 +1,253 @@
+	}
+	if f.Workspace != nil {
+		cmd.Flags().StringVarP(f.Workspace, "workspace", "", "", i18n.T("The workspace name"))
+	}
+	if f.Backend != nil {
+		cmd.Flags().StringVarP(f.Backend, "backend", "", "", i18n.T("The backend to use, supports 'local', 'oss' and 's3'"))
+	}
+	cmd.Flags().StringVarP(&f.Output, "output", "o", f.Output, i18n.T("Specify the output format"))
+}
+
+// ToOptions converts ShowFlags to ShowOptions.
+func (f *ShowFlags) ToOptions() (*ShowOptions, error) {
+	if f.ID == nil || *f.ID == "" {
+		return nil, fmt.Errorf("resource ID is required")
+	}
+
+	var storageBackend backend.Backend
+	var err error
+	if f.Backend != nil && *f.Backend != "" {
+		storageBackend, err = backend.NewBackend(*f.Backend)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		storageBackend, err = backend.NewBackend("")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	workspaceName := ""
+	projectName := ""
+
+	workspaceStorage, err := storageBackend.WorkspaceStorage()
+	if err != nil {
+		return nil, err
+	}
+	if f.Workspace != nil && *f.Workspace != "" {
+		refWorkspace, err := workspaceStorage.Get(*f.Workspace)
+		if err != nil {
+			return nil, err
+		}
+		workspaceName = refWorkspace.Name
+	} else {
+		currentWorkspace, err := workspaceStorage.GetCurrent()
+		if err != nil {
+			return nil, err
+		}
+		workspaceName = currentWorkspace
+	}
+
+	if f.Project != nil && *f.Project != "" {
+		projectName = *f.Project
+	} else {
+		currentProject, _, err := project.DetectProjectAndStacks()
+		if err != nil {
+			return nil, err
+		}
+		projectName = currentProject.Name
+	}
+	storage, err := storageBackend.ReleaseStorage(projectName, workspaceName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ShowOptions{
+		ID:             f.ID,
+		Output:         f.Output,
+		Project:        &projectName,
+		Workspace:      &workspaceName,
+		ReleaseStorage: storage,
+	}, nil
+}
+
+// Validate checks the provided options for the `kusion resource show` command.
+func (o *ShowOptions) Validate(cmd *cobra.Command, args []string) error {
+	if len(args) != 0 {
+		return cmdutil.UsageErrorf(cmd, "Unexpected args: %v", args)
+	}
+
+	return nil
+}
+
+// Run executes the `kusion resource show` command.
+func (o *ShowOptions) Run() (err error) {
+	rel, err := o.ReleaseStorage.Get(o.ReleaseStorage.GetLatestRevision())
+	if err != nil {
+		return fmt.Errorf("no release found for project: %s, workspace: %s", *o.Project, *o.Workspace)
+	}
+
+	if rel.Spec.Resources == nil || len(rel.Spec.Resources) == 0 {
+		return fmt.Errorf("no resources found for project: %s, workspace: %s", *o.Project, *o.Workspace)
+	}
+	resourceMap := make(map[string]apiv1.Resource)
+	for _, res := range rel.Spec.Resources {
+		resourceMap[res.ResourceKey()] = res
+	}
+
+	stateMap := make(map[string]apiv1.Resource)
+	for _, res := range rel.State.Resources {
+		stateMap[res.ResourceKey()] = res
+	}
+
+	if _, ok := resourceMap[*o.ID]; !ok {
```

Comment:
Is it necessary to consider a special corner case here, where a specified resource in the `Release` file is not in the `Spec` but still exists in the `State`? For example, the last `kusion destroy` failed. And if we consider this situation, we might not be able to return an error directly here. cc @Yangyang96

Reply:
> Is it necessary to consider a special corner case here, where a specified resource in the `Release` file is not in the `Spec` but still exists in the `State`? For example, the last `kusion destroy` failed. And if we consider this situation, we might not be able to return an error directly here. cc @Yangyang96

After discussing with @liu-hm19, we have decided to leave the code as it is and set aside this particular corner case. The example above did not sufficiently explain this situation. This case arises when a user attempts to remove a resource using the `kusion apply` operation but it fails, while other commands such as `kusion resource graph` and `kusion destroy` do not handle it differently. This limitation is imposed by the structure of the command-line itself, and `kusion` will not be able to address all of these corner cases. As a result, we have decided to retain your implementation as it accurately reflects the initial user intentions. @hoangndst It's your decision to make to leave it here or change your implementation.

Reply:
For the original purpose of `kusion resource show`, in my opinion, we should keep this approach. 
Thank you for taking the time to provide your comments. 
cc @liu-hm19 @Yangyang96

Reply:
OK 👍 We agree with you. 

And during the discussion about the corner case, we've found a bug concerning the resource elimination in the `State` of the `Release` file when executing the `kusion destroy` command 😂 We will open an issue to track and fix this bug in the future.